packages/lodestar/src/chain/stateCache/stateContextCheckpointsCache.ts

```diff
@@ -83,7 +83,7 @@ export class CheckpointStateCache {
     return previousHits;
   }
 
-  pruneFinalized(finalizedEpoch: Epoch): void {
+  async pruneFinalized(finalizedEpoch: Epoch): Promise<void> {
```

Comment:
made this async so that processFinalizedCheckpoint doesn't need to wait for this to complete

Reply:
Please don't add unnecessary async functions. You can do error handling with try / catch for sync functions